### DISCUSSION TITLE
are sourceip and cat mutually exclusive?

### DISCUSSION THREAD

**Discussion by cagney** [libreswan/libreswan#1072] [Internals]
The code seems to imply that they are, for instance:
```
		address_buf ipb;
		ldbg_sa(child,
			"CAT: configured, not setting host source IP address to %s local CAT = %s->true",
			str_address(&ip, &ipb), bool_str(local->has_cat));
```

**Comment by letoams** [comment]
On Thu, 6 Apr 2023, cagney wrote:

> Subject: [libreswan/libreswan] are sourceip and cat mutually exclusive? (Issue
>      #1071)

> The code seems to imply that they are, for instance:
>
> 		address_buf ipb;
> 		ldbg_sa(child,
> 			"CAT: configured, not setting host source IP address to %s local CAT = %s->true",
> 			str_address(&ip, &ipb), bool_str(local->has_cat));

Yes.
